menuconfig.py

```diff
@@ -299,6 +299,141 @@
     "brightpurple": curses.COLOR_MAGENTA + 8,
 }
 
+def rgb_to_6cube(rgb):
+    """
+    Convert each component of a RGB color to its approximate index
+    in the xterm 6cube color palette
+    """
+    # The formula for converting a single RGB value to the 6cube index
+    rto6 = lambda x: 0 if x < 48 else max(1, int(round((x - 55) / 40)))
+
+    return [rto6(x) for x in rgb]
+
+def rgb_from_6cube(c6):
+    """
+    Convert a 6cube index triplet to the corresponding RGB value
+    """
+    # The formula from converting the xterm 6cube index to a RGB value
+    rfrom6 = lambda x: 0 if x == 0 else 55 + x*40
+    return [rfrom6(x) for x in c6]
+
+def rgb_to_gridx(rgb):
+    """
+    Convert a RGB color to its closest xterm grayscale color
+    (indexes from 0-23 represents xterm colros 232-255)
+    """
+    gray = int(round((rgb[0] + rgb[1] + rgb[2]) / 3))
+
+    # Convert the single gray value component to the xterm greyscale index
+    # (relative to 232)
+    idx = int(round((gray - 8) / 10))
+
+    # Clamp the value between 0 and 23
+    return max(0, min(idx, 23))
+
+def rgb_from_gridx(c):
+    """
+    Convert a grayscale index to its closet single RGB component
+    """
+    g = c * 10 + 8
+    return (g, g, g)
+
+def rgb_alloc(rgb, RGB_COLORS={}):
+    """
+    Ncurses doesn't allow you to define new colors -- you are allowed only
+    to overwrite existing ones.
+
+    The colors from 0-15 are user-defined and there's no way to query the
+    RGB values so we better leave them untouched.
+
+    The RGB values of colors from 16-255 can be easily calculated. However,
+    colors from 232-255 use a different formula (grayscale) so stick with
+    colors from 16-231. 200+ colors should be plentiful.
+    """
+    if rgb in RGB_COLORS:
+        return RGB_COLORS[rgb]
+
+    # The first 16 colors are user defined and we should not change their
+    # values
+    cn = len(RGB_COLORS) + 16
+    if cn >= curses.COLORS:
+        raise StyleError("Unable to allocate new RGB color.")
```

Comment:
`StyleError` no longer exists. Use `_warn()` instead.

A messed-up display is probably nicer than a hard error.